cmd/admin-handlers.go

```diff
@@ -359,31 +368,25 @@ func (a adminAPIHandlers) StartProfilingHandler(w http.ResponseWriter, r *http.R
 	vars := mux.Vars(r)
 	profiler := vars["profilerType"]
 
-	startProfilingResult := make([]StartProfilingResult, len(globalAdminPeers))
-
-	// Call StartProfiling function on all nodes and save results
-	wg := sync.WaitGroup{}
-	for i, peer := range globalAdminPeers {
-		wg.Add(1)
-		go func(idx int, peer adminPeer) {
-			defer wg.Done()
-			result := StartProfilingResult{NodeName: peer.addr}
-			if err := peer.cmdRunner.StartProfiling(profiler); err != nil {
-				result.Error = err.Error()
-				return
-			}
+	var startProfilingResult []StartProfilingResult
+
+	for host, err := range globalNotificationSys.StartProfiling(profiler) {
```

Comment:
Currently, globalNotificationSys.StartProfiling() can be empty if there is no errors with any of the nodes.

Reply:
that is a not a problem if its empty, I am not sure what you are saying.

Reply:
yes this is not a problem `range nil` doesn't crash @vadmeste